metloom/pointdata/base.py

```diff
@@ -242,42 +242,70 @@ def points_from_geometry(
-        """
-        Validate that the GeoDataFrame returned is formatted correctly.
-        The goal of this method is to ensure base classes are returning a
-        consistent format of dataframe
-        """
-        if gdf is None:
-            return
-        assert isinstance(gdf, gpd.GeoDataFrame)
-        columns = gdf.columns
-        index_names = gdf.index.names
-        # check for required indexes
+    def _validate_geodataframe(cls, gdf: gpd.GeoDataFrame):
+        if not isinstance(gdf, gpd.GeoDataFrame):
+            raise DataValidationError('Returned DataFrame must be a GeoDataframe')
+
+    @classmethod
+    def _validate_df_indicies(cls, gdf: gpd.GeoDataFrame):
+        """ Confirm the df is indexed properly"""
         for ei in cls.EXPECTED_INDICES:
-            if ei not in index_names:
+            if ei not in gdf.index.names:
                 raise DataValidationError(
                     f"{ei} was expected, but not found as an"
                     f" index of the final dataframe"
                 )
+
+    @classmethod
+    def _validate_df_columns(cls, gdf: gpd.GeoDataFrame, expected_columns: List[str]):
         # check for expected columns - avoid modifying at class level
-        expected_columns = copy.deepcopy(cls.EXPECTED_COLUMNS)
         possible_extras = ["measurementDate", "quality_code"]
+        columns = gdf.columns
         for pe in possible_extras:
             if pe in columns:
                 expected_columns += [pe]
+
         for column in expected_columns:
             if column not in columns:
                 raise DataValidationError(
                     f"{column} was expected, but not found as a"
                     f" column of the final dataframe"
                 )
 
-        remaining_columns = [c for c in columns if c not in expected_columns]
+    @classmethod
+    def _validate_df_units(cls, gdf: gpd.GeoDataFrame, expected_columns: List[str]):
+        """
+        Check the variables requested have units associated
+        """
+        remaining_columns = [c for c in gdf.columns if c not in expected_columns]
         # make sure all variables have a units column as well
         for rc in remaining_columns:
             if "_units" not in rc:
-                assert f"{rc}_units" in remaining_columns
+                if f"{rc}_units" not in remaining_columns:
+                    raise DataValidationError(f'Missing units column for {rc}')
+
+    @classmethod
+    def validate_sensor_df(cls, gdf: gpd.GeoDataFrame):
+        """
+        Validate that the GeoDataFrame returned is formatted correctly.
+        The goal of this method is to ensure base classes are returning a
+        consistent format of dataframe
+        """
+        if gdf is None:
+            return
+
+        expected_columns = copy.deepcopy(cls.EXPECTED_COLUMNS)
+
+        # Confirm the dataframe is a geodataframe
+        cls._validate_geodataframe(gdf)
```

Comment:
@micah-prime before we push this through I want to make sure you see these new methods for validating the dataframes. I most took you code and broke it out in to functions.

Reply:
This is great! The only issue I see is that _validate_df_columns is called twice